packages/components/src/components/single-select/shadow.tsx

```diff
@@ -0,0 +1,704 @@
+	LabelWithExpertSlotPropType,
+	MsgPropType,
+	NamePropType,
+	SingleSelectAPI,
+	SingleSelectStates,
+	Stringified,
+	SyncValueBySelectorPropType,
+	TooltipAlignPropType,
+	OptionsWithOptgroupPropType,
+	Option,
+} from '../../schema';
+import { Component, Element, h, Host, Method, Prop, State, Watch, Fragment, Listen } from '@stencil/core';
+
+import { nonce } from '../../utils/dev.utils';
+import { stopPropagation, tryToDispatchKoliBriEvent } from '../../utils/events';
+import { SingleSelectController } from './controller';
+
+import type { JSX } from '@stencil/core';
+import { KolButtonWcTag, KolIconTag, KolInputWcTag } from '../../core/component-names';
+import { propagateFocus, showExpertSlot } from '../../schema';
+import { InternalUnderlinedAccessKey } from '../span/InternalUnderlinedAccessKey';
+import { getRenderStates } from '../input/controller';
+import { translate } from '../../i18n';
+
+/**
+ * @slot - Die Beschriftung des Eingabefeldes.
+ */
+@Component({
+	tag: 'kol-single-select',
+	styleUrls: {
+		default: './style.scss',
+	},
+	shadow: true,
+})
+export class KolSingleSelect implements SingleSelectAPI {
+	@Element() private readonly host?: HTMLKolSingleSelectElement;
+	private ref?: HTMLInputElement;
+	private refOptions: HTMLLIElement[] = [];
+	private refScrollContainer?: HTMLUListElement;
+	private oldValue?: string;
+
+	@Method()
+	// eslint-disable-next-line @typescript-eslint/require-await
+	public async getValue(): Promise<string | undefined> {
+		return this.state._value;
+	}
+	private readonly catchRef = (ref?: HTMLInputElement) => {
+		this.ref = ref;
+		propagateFocus(this.host, this.ref);
+	};
+
+	private toggleListbox = (event?: Event) => {
+		if (event) event.preventDefault();
+
+		if (this.state._disabled) {
+			return;
+		} else {
+			this._isOpen = !this._isOpen;
+			if (this._isOpen) {
+				const filterInput = document.getElementById('filter-input');
+				if (filterInput instanceof HTMLInputElement) {
+					filterInput.focus();
+					filterInput.setSelectionRange(0, 0);
+				}
+				this.setFilteredOptionsByQuery(this._inputValue);
+			}
+		}
+	};
+
+	private handleInputValue() {
+		if (Array.isArray(this.state._options) && this.state._options.length > 0 && !this.state._options.some((option) => option.label === this._inputValue)) {
+			this._inputValue = Array.isArray(this._options)
+				? (this.state._options.find((option) => (option as Option<string>).value === this.state._value)?.label as string)
+				: '';
+		}
+	}
+
+	private clearSelection() {
+		if (this.state._disabled) {
+			return;
+		} else {
+			this._focusedOptionIndex = -1;
+			this.state._value = '';
+			this._value = '';
+			this._inputValue = '';
+			this.state._hasValue = false;
+			this._filteredOptions = [...this.state._options];
+		}
+	}
+
+	private selectOption(option: Option<string>) {
+		this.state._value = option.value;
+		this._value = option.value;
+		this._inputValue = option.label as string;
+
+		this.state._hasValue = !!option;
```

Comment:
Muss das tatsächlich ein State sein oder könnten wir eine einfache Methode bzw. [Getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) verwenden, die sich von `this.state._value` ableitet?